src/blitzly/plots/scatter.py

```diff
@@ -1,12 +1,166 @@
+            StandardScaler, MinMaxScaler.
+        size (Optional[Tuple[int, int]): Size of the full plot.
+        show_legend (Optional[bool]): Whether to show the legend.
+        show (bool): Whether to show the figure.
+        write_html_path (Optional[str]): The path to which the histogram should be written as an HTML file.
+            If None, the histogram will not be saved.
+
+    """
+
+    func_list = [
+        "NMF",
+        "PCA",
+        "IncrementalPCA",
+        "KernelPCA",
+        "MiniBatchSparsePCA",
+        "SparsePCA",
+        "TruncatedSVD",
+        "TSNE",
+    ]
+
+    if isinstance(reduction_funcs, str):
+        reduction_funcs = [reduction_funcs]
+
+    if scaler_func not in ["StandardScaler", "MinMaxScaler"]:
+        raise ValueError(
+            f"scaler_func must be one of ['StandardScaler', 'MinMaxScaler]! `{scaler_func}` not supported."
+        )
+
+    if n_components == 3 and len(reduction_funcs) > 1:
+        raise Warning(
+            "Cannot plot more than one plot in 3D! Please either set `n_components` to 2 or `reduction_funcs` to one function."
+        )
+
+    df = check_data(
+        data, min_rows=1, min_columns=1, as_pandas=True, only_numerical_values=False
+    )
+
+    numerical_df = data.select_dtypes(include=[int, float])
+    scaler = getattr(sk_pre, scaler_func)()
+    numerical_df = scaler.fit_transform(numerical_df)
+
+    plots = []
+    for func in set(reduction_funcs):
+
+        if func not in func_list:
+            raise ValueError(
+                f"reduction_func must be one of {func_list}! `{func}` not supported."
+            )
+
+        if func == "TSNE":
+            method: Callable = TSNE
+        else:
+            method: Callable = getattr(sk_decomp, func)  # type: ignore
+
+        red_func = method(n_components=n_components, **reduction_func_kwargs or {})
+        projections = red_func.fit_transform(numerical_df)
+
+        if n_components == 2:
+            fig = px.scatter(
+                projections,
+                x=0,
+                y=1,
+                color=df[target_column],
+                labels={"color": target_column},
+                **plotly_kwargs or {},
+            )
+        elif n_components == 3:
+            fig = px.scatter_3d(
+                projections,
+                x=0,
+                y=1,
+                z=2,
+                color=df[target_column],
+                labels={"color": target_column},
+                **plotly_kwargs or {},
+            )
+        else:
+            raise ValueError("`n_components` must be 2 or 3!")
+
+        plots.append(update_figure_layout(fig, title + f" ({func})", size=None))
+
+    len_plots = len(plots)
+    if len_plots == 1:
+        shape = (1, 1)
+    elif len_plots % 2 == 0:
+        shape = (len_plots // 2, 2)
+    else:
+        shape = (len_plots // 2 + 1, 2)
```

Comment:
```suggestion
shape = (int(np.ceil(len_plots / 2)), min(len_plots, 2))
```

You could remove the entire if-else block and add the suggested line.

Reply:
Nice improvement. Thanks. I changed the implementation.